ml_frameworks_bot/keras_io/retriever.py

```diff
@@ -56,13 +62,20 @@ def from_wandb_artifact(cls, artifact_address: str, similarity_top_k: int = 10):
         vector_index = load_index_from_storage(
             storage_context=StorageContext.from_defaults(persist_dir=artifact_dir)
         )
-        return cls(
+        _cls = cls(
             embedding_model_name=embedding_model_name,
             similarity_top_k=similarity_top_k,
             vector_index=vector_index,
         )
+        _cls.wandb_artifact_address = artifact_address
+        return _cls
 
-    def load_documents(self, num_workers: Optional[int] = None) -> List[Document]:
+    def load_documents(
+        self,
+        included_directories: List[str] = ["examples", "guides", "templates"],
+        exclude_file_postfixes: List[str] = ["index.md"],
+        return_nodes: bool = True,
```

Comment:
Question: why the `return_nodes` parameter? Shouldn't we just do text nodes for a single reference?

Reply:
I left an optional flag `apply_chunking` in `KerasIORetreiver.index_documents`, which optionally applies semantic chunking to the documents before generating the embeddings. When `apply_chunking=True`, we need `Document`s instead of `TextNode`s.